### PR TITLE
refactor, fix, chore(Egen): changes and refactors the get_started_vertex_training_xgboost.ipynb notebook

### PR DESCRIPTION
**Pull request by k-root**
The following changes are made in the notebook:
- Fixes argument passed in pip install
- updates packages
- removes boilerplate code
- adds link to colab enterprise
- updates usage of 'will' appropriately
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb [hunk 1]
```diff
@@ -8,7 +8,7 @@
       },
       "outputs": [],
       "source": [
-        "# Copyright 2022 Google LLC\n",
+        "# Copyright 2024 Google LLC\n",
```
Comment (katiemn) [inline notebooks/official/custom/get_started_vertex_training_xgboost.ipynb:11]: The year can remain 2022

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb [hunk 2]
```diff
@@ -348,43 +319,7 @@
...
-    {
-      "cell_type": "code",
-      "execution_count": null,
-      "metadata": {
-        "id": "JZg2sszQox9T"
-      },
-      "outputs": [],
-      "source": [
-        "aiplatform.init(project=PROJECT_ID, staging_bucket=BUCKET_URI)"
+        "Next, set up some variables used throughout the tutorial."
```
Comment (katiemn) [inline notebooks/official/custom/get_started_vertex_training_xgboost.ipynb:322]: Can you please change 'used' to 'to use'?

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb [hunk 3]
```diff
@@ -406,7 +341,7 @@
         "\n",
         "Learn more about [hardware accelerator support for your region](https://cloud.google.com/vertex-ai/docs/general/locations#accelerators).\n",
         "\n",
-        "*Note*: TF releases before 2.3 for GPU support will fail to load the custom model in this tutorial. It is a known issue and fixed in TF 2.3. This is caused by static graph ops that are generated in the serving function. If you encounter this issue on your own custom models, use a container image for TF 2.3 with GPU support."
+        "*Note*: TF releases before 2.3 for GPU support is expected to fail to load the custom model in this tutorial. It is a known issue and fixed in TF 2.3. This is caused by static graph ops that are generated in the serving function. If you encounter this issue on your own custom models, use a container image for TF 2.3 with GPU support."
```
Comment (kittyabs) [inline notebooks/official/custom/get_started_vertex_training_xgboost.ipynb:344]: change "It is" to "It's" (i.e., contraction)

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb [hunk 4]
```diff
@@ -503,8 +438,8 @@
       "source": [
         "## Introduction to XGBoost training\n",
         "\n",
-        "Once you have trained a XGBoost model, you will want to save it at a Cloud Storage location, so it can subsequently be uploaded to a `Vertex AI Model` resource.\n",
-        "The XGBoost package does not have support to save the model to a Cloud Storage location. Instead, you will do the following steps to save to a Cloud Storage location.\n",
+        "Once you have trained a XGBoost model, save it at a Cloud Storage location, so it can subsequently be uploaded to a `Vertex AI Model` resource.\n",
```
Comment (kittyabs) [inline notebooks/official/custom/get_started_vertex_training_xgboost.ipynb:441]: remove back tics. Should be Vertex AI model" (where "model" is lower case)

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb [hunk 5]
```diff
@@ -534,11 +476,16 @@
         "\n",
         "The files `setup.cfg` and `setup.py` are the instructions for installing the package into the operating environment of the Docker image.\n",
         "\n",
-        "The file `trainer/task.py` is the Python script for executing the custom training job. *Note*, when we referred to it in the worker pool specification, we replace the directory slash with a dot (`trainer.task`) and dropped the file suffix (`.py`).\n",
-        "\n",
-        "#### Package Assembly\n",
-        "\n",
-        "In the following cells, you will assemble the training package."
+        "The file `trainer/task.py` is the Python script for executing the custom training job. *Note*, when we referred to it in the worker pool specification, we replace the directory slash with a dot (`trainer.task`) and dropped the file suffix (`.py`).\n"
```
Comment (kittyabs) [inline notebooks/official/custom/get_started_vertex_training_xgboost.ipynb:479]: change note to remove "we":
*Note*, when `trainer/task.py` is referred to in the worker pool specification, the directory slash is replaced with a dot and the file suffix (`.py`) suffix is dropped: (`trainer.task`).

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb [hunk 6]
```diff
@@ -951,14 +898,14 @@
         "id": "cleanup:mbsdk"
       },
       "source": [
-        "# Cleaning up\n",
-        "\n",
+        "## Cleaning up\n",
```
Comment (kittyabs) [inline notebooks/official/custom/get_started_vertex_training_xgboost.ipynb:901]: Change "Cleaning up" to "Clean up"

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb [hunk 7]
```diff
@@ -951,14 +898,14 @@
...
         "To clean up all Google Cloud resources used in this project, you can [delete the Google Cloud\n",
         "project](https://cloud.google.com/resource-manager/docs/creating-managing-projects#shutting_down_projects) you used for the tutorial.\n",
         "\n",
         "Otherwise, you can delete the individual resources you created in this tutorial:\n",
         "\n",
-        "- Custom Job (Custome Training job is remove in previous step)\n",
+        "- Custom Job (Custom Training job is removed in previous step)\n",
```
Comment (kittyabs) [inline notebooks/official/custom/get_started_vertex_training_xgboost.ipynb:907]: add "the":
"Custom Job (Custom Training job is removed in the previous step)"

**Review by kittyabs — COMMENTED** [review]
Some suggested edits

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb (hunk 7 above)
Comment (k-root) [reply notebooks/official/custom/get_started_vertex_training_xgboost.ipynb]: added the change

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb (hunk 6 above)
Comment (k-root) [reply notebooks/official/custom/get_started_vertex_training_xgboost.ipynb]: rectified

File: notebooks/official/custom/get_started_vertex_training_xgboost.ipynb (hunk 5 above)
Comment (k-root) [reply notebooks/official/custom/get_started_vertex_training_xgboost.ipynb]: Rectified

**Pull request merged**